src/averageAges.js

```diff
@@ -55,7 +75,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let averageAgeDiff = 0;
+  let children = people.filter(
+    child => people.find(mom => (mom.name === child.mother))
+  );
+
+  children = onlyWithSon
+    ? children.filter(child => child.sex === 'm')
+    : children;
+
+  const womenAges = children
+    .map(child => (
+      child.born - (people.find(mom => mom.name === child.mother).born)
+    ));
+
+  averageAgeDiff = womenAges
+    .reduce((accumulator, currentValue) =>
+      accumulator + currentValue) / womenAges.length;
+
+  return averageAgeDiff;
```

Comment:
You can simply return needed value here.

```suggestion
  return womenAges
    .reduce((accumulator, currentValue) =>
      accumulator + currentValue) / womenAges.length;
```